aiodocker/multiplexed.py

```diff
@@ -1,40 +1,78 @@
 import asyncio
 import struct
-
 import aiohttp
 
 from . import constants
 
+# aiohttp has no errors.py
+try:
+    import aiohttp.errors as errors
+except:
+    import aiohttp.client_exceptions as errors
+
 
 class MultiplexedResult:
     def __init__(self, response):
         self.response = response
 
     async def fetch(self):
-        while True:
-            try:
-                hdrlen = constants.STREAM_HEADER_SIZE_BYTES
-                header = await self.response.content.readexactly(hdrlen)
-                _, length = struct.unpack('>BxxxL', header)
-                if not length:
-                    continue
-                data = await self.response.content.readexactly(length)
-            except (aiohttp.ClientConnectionError,
-                    aiohttp.ServerDisconnectedError):
-                break
-            except asyncio.IncompleteReadError:
-                break
-            yield data.decode('utf8')
+        try:
+            while True:
+                try:
+                    hdrlen = constants.STREAM_HEADER_SIZE_BYTES
+                    header = await self.response.content.readexactly(hdrlen)
+                    _, length = struct.unpack('>BxxxL', header)
+                    if not length:
+                        continue
+                    data = await self.response.content.readexactly(length)
+                #except (aiohttp.errors.ClientDisconnectedError,
+                #        aiohttp.errors.ServerDisconnectedError):
+                except errors.ServerDisconnectedError:
+                    break
+                except asyncio.IncompleteReadError:
+                    break
+                yield data.decode('utf8')
+        finally:
+
+            await self.close()
+
+    async def fetch_raw(self):
+        try:
+            async for data in self.response.content.iter_chunked(1024):
+                yield data.decode('utf8')
```

Comment:
As this is "raw" result, I think it's better NOT to decode using arbitrary encoding here.
Just return the bytes as-is.